ngraph_bridge/enable_variable_ops/ngraph_variable_update_ng_tensor_op.cc

```diff
@@ -0,0 +1,117 @@
+#include "ngraph_bridge/ngraph_timer.h"
+#include "ngraph_bridge/ngraph_utils.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+//---------------------------------------------------------------------------
+//  NGraphVariableUpdateNGTensorOp::ctor
+//---------------------------------------------------------------------------
+NGraphVariableUpdateNGTensorOp::NGraphVariableUpdateNGTensorOp(
+    OpKernelConstruction* context)
+    : OpKernel(context) {
+  OP_REQUIRES_OK(context, context->GetAttr("ngraph_graph_id", &ng_graph_id_));
+  OP_REQUIRES_OK(context, context->GetAttr("ngraph_variable_shared_name",
+                                           &ng_variable_shared_name_));
+
+  NGRAPH_VLOG(4) << "NGraphVariableUpdateNGTensorOp:: Constructor called for: "
+                 << def().name() << " ,Graph ID " << ng_graph_id_
+                 << " ngraph_variable_shared_name " << ng_variable_shared_name_;
+
+  OP_REQUIRES(context, IsRefType(context->input_type(0)),
+              errors::InvalidArgument("input[0] needs to be a ref type"));
+}
+
+//---------------------------------------------------------------------------
+//  ~NGraphVariableUpdateNGTensorOp
+//---------------------------------------------------------------------------
+NGraphVariableUpdateNGTensorOp::~NGraphVariableUpdateNGTensorOp() {
+  NGRAPH_VLOG(4) << "~NGraphVariableUpdateNGTensorOp::" << name() << endl;
+}
+
+//---------------------------------------------------------------------------
+// OpKernel::Compute
+//---------------------------------------------------------------------------
+void NGraphVariableUpdateNGTensorOp::Compute(OpKernelContext* context) {
+  std::ostringstream oss;
+  // Start event tracing
+  ngraph::Event event_compute(oss.str(), name(), "");
+  bool log_copies = false;
+  OP_REQUIRES_OK(context,
+                 IsNgraphTFLogTensorCopiesEnabled(ng_graph_id_, log_copies));
+  std::stringstream copy_log_str;
+  NGRAPH_VLOG(4) << "KERNEL[" << type_string() << "]: " << name() << "\n";
+  int number_of_copies = 0;
+  NGRAPH_VLOG(4) << "NGraphVariableUpdateNGTensorOp:: Compute called for: "
+                 << def().name() << " ,Graph ID " << ng_graph_id_ << "\n";
+
+  // Since we have ngraph_variable_shared_name as an attribute,
+  // we can use that to get the variable from the context
+  NGraphVar* var;
+
+  OP_REQUIRES_OK(context, context->resource_manager()->Lookup<NGraphVar>(
+                              context->resource_manager()->default_container(),
+                              ng_variable_shared_name_, &var));
+
+  // Set the output Ref Tensor at output_index to be an alias of the
+  // input Ref Tensor at input_index.
+  // REQUIRES: IsRefType(input_dtype(input_index)).
+  // REQUIRES: IsRefType(output_dtype(output_index)).
+  context->forward_ref_input_to_ref_output(0, 0);
+
+  if (var->copy_tf_to_ng()) {
+    number_of_copies++;
```

Comment:
`number_of_copies` seems to be a local variable. Which means it is either going to be 0 or 1. Maybe this was not the intention?